linera-base/src/data_types.rs

```diff
@@ -880,20 +875,53 @@ pub struct CompressedBytecode {
     pub compressed_bytes: Vec<u8>,
 }
 
+#[cfg(not(target_arch = "wasm32"))]
 impl CompressedBytecode {
+    /// Returns `true` if the decompressed size does not exceed the limit.
+    pub fn decompressed_size_at_most(&self, limit: u64) -> Result<bool, DecompressionError> {
+        let mut decoder = zstd::stream::Decoder::new(&*self.compressed_bytes)?;
+        let limit = usize::try_from(limit).unwrap_or(usize::MAX);
```

Comment:
Maybe pass in the `limit` as `usize` type? We already use `usize::MAX` as the fallback when limit is not specificed. It seems fishy that we use the same constant for two different reasons.

Reply:
We have this problem in multiple places: Network-wide config values can't be architecture-dependent.

My view is that we should avoid `usize` as much as possible and only do the conversion close to the place where we need it.

Eventually we should clearly define which architectures are supported; then all limits for anything that ends up in a vector (or slice) must have a limit that is small enough for a vector in those architectures.